nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/blob/BlobToPDF.java

```diff
@@ -79,26 +93,46 @@ public Blob run(DocumentModel doc) {
     }
 
     @OperationMethod
-    public Blob run(Blob blob) {
-        if ("application/pdf".equals(blob.getMimeType())) {
+    public Blob run(Blob blob) throws IOException {
+        String mimetype = blob.getMimeType();
+        if ("application/pdf".equals(mimetype)) {
             return blob;
         }
-        BlobHolder bh = new SimpleBlobHolder(blob);
-        bh = service.convertToMimeType("application/pdf", bh, new HashMap<String, Serializable>());
-        Blob result = bh.getBlob();
+        Blob result;
+        if (blob instanceof StringBlob && !"text/plain".equals(mimetype)) {
```

Comment:
Because in the `replaceURLsByAbsolutePaths` method, I retrieve `blob.getString()` the I replace the blob by a new `StringBlob` with the new String content.
What do you think?